corehq/ex-submodules/dimagi/utils/dates.py

```diff
@@ -123,40 +123,16 @@ def __init__(self, startdate, enddate, format=ISO_DATE_FORMAT, inclusive=True, t
         self.timezone = timezone
         self.max_days = max_days
 
-    def __getstate__(self):
-        """
-            For pickling the DateSpan object.
-        """
-        return dict(
-            startdate=self.startdate.isoformat() if self.startdate else None,
-            enddate=self.enddate.isoformat() if self.enddate else None,
-            format=self.format,
-            inclusive=self.inclusive,
-            is_default=self.is_default,
-            timezone=self.timezone.zone,
-            max_days=self.max_days,
+    def __eq__(self, other):
+        return (
+            self.startdate == other.startdate
+            and self.enddate == other.enddate
+            and self.format == other.format
+            and self.inclusive == other.inclusive
+            and self.timezone == other.timezone
+            and self.max_days == other.max_days
         )
 
-    def __setstate__(self, state):
-        """
-            For un-pickling the DateSpan object.
-        """
-        logging = get_task_logger(__name__) # logging is likely to happen within celery
-        try:
-            self.startdate = dateutil.parser.parse(state.get('startdate')) if state.get('startdate') else None
-            self.enddate = dateutil.parser.parse(state.get('enddate')) if state.get('enddate') else None
-        except Exception as e:
-            logging.error("Could not unpack start and end dates for DateSpan. Error: %s" % e)
-        self.format = state.get('format', ISO_DATE_FORMAT)
-        self.inclusive = state.get('inclusive', True)
-        self.timezone = pytz.utc
-        self.is_default = state.get('is_default', False)
-        self.max_days = state.get('max_days')
-        try:
-            self.timezone = pytz.timezone(state.get('timezone'))
-        except Exception as e:
-            logging.error("Could not unpack timezone for DateSpan. Error: %s" % e)
-
```

Comment:
Does anyone know why we have this weird custom pickling code and/or whether we for some reason still need it?

Reply:
@biyeun any ideas? 🤣 https://github.com/dimagi/dimagi-utils/commit/20c31c86700ba07719618daf209cd0de4b888b06

Reply:
I'm guessing there was an issue with a previous pickling protocol or something like that that we had to work around